src/syrupy/assertion.py

```diff
@@ -300,6 +307,11 @@ def _assert(self, data: "SerializableData") -> bool:
         matches = False
         assertion_success = False
         assertion_exception = None
+        matcher_options = None
+        for key,value in self._extra_args.items():
+            if key == "matcher_options":
+                matcher_options = value
```

Comment:
This enforces a certain schema in the extra_args object which if we want to support arbitrary args we shouldn't do. Instead, we should propagate the "extra_args" to all relevant methods.

Reply:
Removed this. I also forwaded `extra_args` to different functions. Could you please tell me  what you think about this?